src/main/java/com/pashonokk/dvdrental/controller/FilmController.java

```diff
@@ -0,0 +1,70 @@
+package com.pashonokk.dvdrental.controller;
+
+import com.pashonokk.dvdrental.dto.CategoryDto;
+import com.pashonokk.dvdrental.dto.FilmDto;
+import com.pashonokk.dvdrental.dto.FilmSavingDto;
+import com.pashonokk.dvdrental.dto.PageDto;
+import com.pashonokk.dvdrental.exception.BigSizeException;
+import com.pashonokk.dvdrental.service.FilmService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/films")
+public class FilmController {
+    private final FilmService filmService;
+
+    @GetMapping
+    public ResponseEntity<Page<FilmDto>> getFilms(@RequestBody PageDto pageDto) {
+        if (pageDto.getSize() > 100) {
+            throw new BigSizeException("You can get maximum 100 films at one time");
+        }
+        Page<FilmDto> allFilms = filmService.getAllFilms(PageRequest.of(pageDto.getPage(), pageDto.getSize(), Sort.by(pageDto.getSort())));
+        return ResponseEntity.ok(allFilms);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<FilmDto> getFilmById(@PathVariable Long id) {
+        FilmDto filmDto = filmService.getFilmWithCategories(id);
+        return ResponseEntity.ok(filmDto);
+    }
+
+    @GetMapping("/{id}/categories")
+    public ResponseEntity<List<CategoryDto>> getFilmsCategories(@PathVariable Long id) {
+        List<CategoryDto> filmsCategories = filmService.getFilmsCategories(id);
+        return ResponseEntity.ok(filmsCategories);
+    }
+
+    @PostMapping
+    public ResponseEntity<FilmDto> addFilm(@RequestBody FilmSavingDto filmSavingDto) {
+        FilmDto savedFilm = filmService.addFilm(filmSavingDto);
+        return ResponseEntity.created(URI.create("localhost:10000/films/" + savedFilm.getId())).body(savedFilm);
```

Comment:
Be careful with hardcoding "localhost:10000" in your application. This might not work when you deploy your app to different environments. Consider using ServletUriComponentsBuilder or store the base URL in a configuration file.